Unable to compile v6.0.0 on windows

I wasn't able to compile latest version on windows. This seems related to the GHC upgrade we had recently.

- Error for GHC 8.4.4: https://ci.appveyor.com/project/postgrest/postgrest/builds/25457530
- Error for GHC 8.6.5: https://ci.appveyor.com/project/steve-chavez/postgrest/builds/25506383 

These errors are on AppVeyor, but I also hit the same errors when trying to compile on a Windows Server 2019 in an ec2 instance.

Any help would be greatly appreciated. Trying to [build from source](http://postgrest.org/en/v5.2/install.html#build-from-source) on Windows and reporting here would help.

@robx Both hang.

`windows-reduced` hangs with:
```
src\Integral.hs:106:1: warning:
    Forall'd constraint `Ord ▒' is not bound in RULE lhs
      Orig bndrs: [▒, p, $dOrd_a4Cz, $dIntegral_a4CA, $dPrinter_a4CB]
      Orig lhs: let {
                  $dPrinter_a4CI :: Printer p
                  [LclId]
                  $dPrinter_a4CI = $dPrinter_a4CB } in
                let {
                  $dIntegral_a4CH :: Integral ▒
                  [LclId]
                  $dIntegral_a4CH = $dIntegral_a4CA } in
                let {
                  $dPositionalSystem_a4CF :: PositionalSystem Binary
                  [LclId]
                  $dPositionalSystem_a4CF = Integral.$fPositionalSystemBinary } in
                let {
                  $dReal_a5aX :: Real ▒
                  [LclId]
                  $dReal_a5aX = GHC.Real.$p1Integral @ ▒ $dIntegral_a4CA } in
                let {
                  $dOrd_a5aZ :: Ord ▒
                  [LclId]
                  $dOrd_a5aZ = GHC.Real.$p2Real @ ▒ $dReal_a5aX } in
                let {
                  $dOrd_a4CG :: Ord ▒
                  [LclId]
                  $dOrd_a4CG = $dOrd_a5aZ } in
                number'
                  @ Binary
                  @ ▒
                  @ p
                  $dPositionalSystem_a4CF
                  $dOrd_a4CG
                  $dIntegral_a4CH
                  $dPrinter_a4CI
      optimised lhs: number'
                       @ Binary
                       @ ▒
                       @ p
                       $dPositionalSystem_a4CF
                       $dOrd_a4CG
                       $dIntegral_a4CA
                       $dPrinter_a4CB
```

For `windows-buffer-error`:

```
Building library for text-printer-0.5..
[1 of 1] Compiling Integral

src\Integral.hs:79:1: warning:
    Forall'd constraint `Ord ▒' is not bound in RULE lhs
      Orig bndrs: [▒, p, $dOrd_a4ze, $dIntegral_a4zf, $dPrinter_a4zg]
      Orig lhs: let {
                  $dPrinter_a4zn :: Printer p
                  [LclId]
                  $dPrinter_a4zn = $dPrinter_a4zg } in
                let {
                  $dIntegral_a4zm :: Integral ▒
                  [LclId]
                  $dIntegral_a4zm = $dIntegral_a4zf } in
                let {
                  $dPositionalSystem_a4zk :: PositionalSystem Binary
                  [LclId]
                  $dPositionalSystem_a4zk = Integral.$fPositionalSystemBinary } in
                let {
                  $dReal_a4X9 :: Real ▒
                  [LclId]
                  $dReal_a4X9 = GHC.Real.$p1Integral @ ▒ $dIntegral_a4zf } in
                let {
                  $dOrd_a4Xb :: Ord ▒
                  [LclId]
                  $dOrd_a4Xb = GHC.Real.$p2Real @ ▒ $dReal_a4X9 } in
                let {
                  $dOrd_a4zl :: Ord ▒
                  [LclId]
                  $dOrd_a4zl = $dOrd_a4Xb } in
                number'
                  @ Binary
                  @ ▒
                  @ p
                  $dPositionalSystem_a4zk
                  $dOrd_a4zl
                  $dIntegral_a4zm
                  $dPrinter_a4zn
      optimised lhs: number'
                       @ Binary
                       @ ▒
                       @ p
                       $dPositionalSystem_a4zk
                       $dOrd_a4zl
                       $dIntegral_a4zf
                       $dPrinter_a4zg
   |
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,10 +1,5 @@
-## Travis is used for building an OSX binary on new tag releases,
-## it also compiles the project on OSX for each commit to master.
-##
-## Building the whole project can take longer than 50 minutes. Since Travis has a global timeout of 50 minutes
-## we compile for 30 minutes tops(see `gtimeout 1800` below) and quit compiling with no error.
-## Since we CACHE the compile results we can continue compiling from where we left off
-## on the next commit.
+## Travis is only used for building an OSX binary ,
+## no tests are run here.
 language: generic
 
 sudo: false
@@ -47,6 +42,10 @@ install:
   fi
 
 script:
+## Building the whole project can take longer than 50 minutes. Since Travis has a global timeout of 50 minutes
+## we compile for 30 minutes tops(`gtimeout 1800`) and quit compiling with no error.
+## Since we CACHE the compile results we can continue compiling from where we left off
+## on the next commit.
 - gtimeout 1800 stack build --no-terminal --only-snapshot --install-ghc || true
 - |
   if test ! "$TRAVIS_TAG"

--- a/.travis.yml
+++ b/.travis.yml
@@ -51,7 +51,7 @@ script:
 - |
   if test ! "$TRAVIS_TAG"
   then
-    echo 'No tag pushed. Skipping build.'
+    echo 'No tag pushed. Skip building binary.'
   else
     stack build --no-terminal --copy-bins --local-bin-path .
   fi

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -24,7 +24,7 @@ install:
 build_script:
 - stack setup --no-terminal > nul
 # Appveyor has a timeout of 60 mins, building can take longer, limit the time and make sure this succeeds,
-# previous work will get cached and finish on next commit.
+# previous work will get cached and finish on next commit
 - bash -lc "timeout 2700 'C:\projects\postgrest\stack.exe' build -j1 --copy-bins --local-bin-path . || true"
 
 artifacts:

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -36,5 +36,5 @@ artifacts:
 - path: postgrest.exe
 
 deploy_script:
-- 7z a -tzip postgrest-%APPVEYOR_REPO_TAG_NAME%-windows-x64.zip postgrest.exe
-- bash -lc " exec 0</dev/null && test -n \"$APPVEYOR_REPO_TAG_NAME\" && cd $APPVEYOR_BUILD_FOLDER && ghr -t $GITHUB_TOKEN -u $APPVEYOR_ACCOUNT_NAME -r $APPVEYOR_PROJECT_NAME -b \"$(sed -n \"1,/$(echo $APPVEYOR_REPO_TAG_NAME | cut -c2-)/d;/## \[/q;p\" CHANGELOG.md)\" --replace $APPVEYOR_REPO_TAG_NAME postgrest-$APPVEYOR_REPO_TAG_NAME-windows-x64.zip"
+- IF DEFINED %APPVEYOR_REPO_TAG_NAME% 7z a -tzip postgrest-%APPVEYOR_REPO_TAG_NAME%-windows-x64.zip postgrest.exe
+- IF DEFINED %APPVEYOR_REPO_TAG_NAME% bash -lc " exec 0</dev/null && cd $APPVEYOR_BUILD_FOLDER && ghr -t $GITHUB_TOKEN -u $APPVEYOR_ACCOUNT_NAME -r $APPVEYOR_PROJECT_NAME -b \"$(sed -n \"1,/$(echo $APPVEYOR_REPO_TAG_NAME | cut -c2-)/d;/## \[/q;p\" CHANGELOG.md)\" --replace $APPVEYOR_REPO_TAG_NAME postgrest-$APPVEYOR_REPO_TAG_NAME-windows-x64.zip"

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -10,6 +10,7 @@ environment:
   global:
     STACK_ROOT: "c:\\sr"
     GOPATH: c:\gopath
+    TMP: "c:\\tmp"
 
 test: off
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,5 +1,4 @@
-## AppVeyor is only used for building a Windows binary,
-## no tests are run here.
+## AppVeyor is only used for building a Windows binary, no tests are run here.
 platform: x64
 
 cache:

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -10,7 +10,6 @@ environment:
   global:
     STACK_ROOT: "c:\\sr"
     GOPATH: c:\gopath
-    TMP: "c:\\tmp"
 
 test: off
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -14,13 +14,11 @@ environment:
 
 test: off
 
-skip_non_tags: true
-
-skip_branch_with_pr: true
-
-branches:
-  only:
-    - master
+#skip_non_tags: true
+#skip_branch_with_pr: true
+#branches:
+  #only:
+    #- master
 
 install:
 - set PATH=C:\Program Files\PostgreSQL\9.6\bin\;%PATH%
@@ -31,6 +29,7 @@ install:
 
 build_script:
 - stack setup --no-terminal > nul
+# Appveyor has a timeout of 60 mins, building can take longer, limit the time and make sure this succeeds
 - bash -lc "timeout 2700 'C:\projects\postgrest\stack.exe' build -j1 --copy-bins --local-bin-path . || true"
 
 artifacts:
@@ -38,4 +37,4 @@ artifacts:
 
 deploy_script:
 - 7z a -tzip postgrest-%APPVEYOR_REPO_TAG_NAME%-windows-x64.zip postgrest.exe
-- bash -lc "exec 0</dev/null && cd $APPVEYOR_BUILD_FOLDER && ghr -t $GITHUB_TOKEN -u $APPVEYOR_ACCOUNT_NAME -r $APPVEYOR_PROJECT_NAME -b \"$(sed -n \"1,/$(echo $APPVEYOR_REPO_TAG_NAME | cut -c2-)/d;/## \[/q;p\" CHANGELOG.md)\" --replace $APPVEYOR_REPO_TAG_NAME postgrest-$APPVEYOR_REPO_TAG_NAME-windows-x64.zip"
+- bash -lc " exec 0</dev/null && test -n \"$APPVEYOR_REPO_TAG_NAME\" && cd $APPVEYOR_BUILD_FOLDER && ghr -t $GITHUB_TOKEN -u $APPVEYOR_ACCOUNT_NAME -r $APPVEYOR_PROJECT_NAME -b \"$(sed -n \"1,/$(echo $APPVEYOR_REPO_TAG_NAME | cut -c2-)/d;/## \[/q;p\" CHANGELOG.md)\" --replace $APPVEYOR_REPO_TAG_NAME postgrest-$APPVEYOR_REPO_TAG_NAME-windows-x64.zip"

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,4 +1,5 @@
-## AppVeyor is used for building a Windows binary on new tag releases
+## AppVeyor is only used for building a Windows binary,
+## no tests are run here.
 platform: x64
 
 cache:
@@ -14,12 +15,6 @@ environment:
 
 test: off
 
-#skip_non_tags: true
-#skip_branch_with_pr: true
-#branches:
-  #only:
-    #- master
-
 install:
 - set PATH=C:\Program Files\PostgreSQL\9.6\bin\;%PATH%
 - curl -sS -ostack.zip -L --insecure http://www.stackage.org/stack/windows-x86_64
@@ -29,7 +24,8 @@ install:
 
 build_script:
 - stack setup --no-terminal > nul
-# Appveyor has a timeout of 60 mins, building can take longer, limit the time and make sure this succeeds
+# Appveyor has a timeout of 60 mins, building can take longer, limit the time and make sure this succeeds,
+# previous work will get cached and finish on next commit.
 - bash -lc "timeout 2700 'C:\projects\postgrest\stack.exe' build -j1 --copy-bins --local-bin-path . || true"
 
 artifacts:

--- a/stack.yaml
+++ b/stack.yaml
@@ -13,6 +13,8 @@ extra-deps:
 - deferred-folds-0.9.10.1
 - primitive-0.6.4.0
 - jose-0.8.1.0
+- text-printer-0.5.0.1
+- network-2.7.0.1
 ghc-options:
   postgrest: -O2 -Werror -Wall -fwarn-identities -fno-warn-redundant-constraints -optP-Wno-nonportable-include-path
 nix:

